### ISSUE TITLE
[Bug] Stacking charts with different curves will cause problems with background color overwriting

### ISSUE THREAD

**Issue by fawdlstty** [apache/echarts#18639]
### Version

5.4.2

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?c=area-stack&code=PYBwLglsB2AEC8sDeAoW7aTAGwKYC5k0MSxcAPMQgckDANQA-VA_tWuPQF8AaVzYYbSEIVQlSAJwgBzCblE0AhuQgBnFiU7c806ABMh3Etrlg5hANrVDxgIzUOsC0bkAmagF1u6khPG6iIjHgAZlT2AMwApLb6GOISABYh1AAskVz-6ABGwGBgwAC2NBFR6bAAxjDGENAAMnIZuNiEYKIArrgeaRjkAIKKSnolYACeIAT2pUa4EsCiQ8XpWS06crMA4nKCsIFy2Eq4nf6WJrDmgBYqgABytvangDFyV9SngJBy96eA22ovgClyL4C2ii-Ap6ZuDrcIa9ZQDdLDUY0ABuOzaqgwngwe3EuH6J2i6GEJQw0DkeTGDmMLgOOMhhOwVVw8xxsCUxlKAGsaIBuA0A84k0nErXByADKwzw4NpJGgMzyOyFwv8oDkpQgw0IAAYAHQAVkxOLYGrUpJK9Nwm2ouB0nJKRzMTicirsVgAHE4bQBOKx2JyhJKux3W2ChULe0JWRXuEpI_zY2l4gk0I42XUQkYUqmm9L02XM-zs5P-bl8gVjJChnHmk5WK020IO2CBl1V912L2uv2uwPB9JakjuNgAbiAA

### Steps to Reproduce

1. open url
2. run

### Current Behavior

Stack two different curves

### Expected Behavior

The background fills correctly

### Environment

```markdown
- OS: win11
- Browser: chrome
- Framework: empty
```

**Comment by plainheart** [comment]
Try to set [`z`](https://echarts.apache.org/option.html#series-line.z) for each series. [Demo](https://echarts.apache.org/examples/editor.html?c=area-stack&code=PYBwLglsB2AEC8sDeAoW7aTAGwKYC5k0MSxcAPMQgckDANQA-VA_tWuPQF8AaVzYYbSEIVQlSAJwgBzCblE0AhuQgBnFiU7c806ABMh3Etrlg5hANrVDxgIzUOsC0bkAmagF1u6khPG6iIjHgAZlT2AMwApLb6GOISABYh1AAskVz-6ABGwGBgwAC2NBFR6bAAxjDGENAAMnIZuNiEYKIArrgeaRjkAIKKSnolYACeIAT2pUa4EsCiQ8XpWS06crMA4nKCsIFy2Eq4nf6WJrDmgBYqgABytvangDFyV9SngJBy96eA22ovgClyL4C2ii-Ap6ZuDrcIa9ZQDdLDUY0ABuOzaqgwngwe3EuH6J2i6GEJQw0DkeTGDmMLgOOMhhOwVVw8xxsCUxlKAGsaIBuA0A84k0nErXByADKwzw4NpJGgMzyOyFwv8oDkpQgw0IAAYAHQAVkxOLYGrUpJK9Nwm2ouB0nJKRzMTicirsVgAHE4bQBOKx2JyhJKux3W2ChULe0JWRXuEpI_zY2l4gk0I42XUQkYUqmm9L02XM-zs5P-bl8gVjJChnHmk5WK020IO2CBl1V912L2uv2uwPB9JakjuNgAbiAA)

**Comment by fawdlstty** [comment]
Z can't solve this problem

**Comment by helgasoft** [comment]
@fawdlstty, for the stacking _areaStyle_ to work, all lines need to be of the same type - straight, or smooth, or step. Your example shows that they cannot be mixed.

As for a pretty picture, this hack will do - [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=area-stack&code=PTAEGMHsDsBcCcCWAjArrRNTIJ6gBYCmANgOYCGAzpAGawB0UAtgFAAmboAvKANoug-AJiEAGADSgAjAA4hkqQE4pkoQGYALKsUTQatbrVTRAXXEC-UsQrXzpoldM2Sdqg6uMmWXyAAcMWDwA3haCGLDEhABcoEGgsIQAHrAxAOSAYBqAB8qAf2qpoAC-5oJhkJDEGL4xIcU18UikpITwaeSJiJSpkqG1NJDwTOSwCc2gABS-AJTcAHyx3bWgiDRjvvSR0KSw-NNCU6uUvsSI4ISjElITANzztUPcgr68pvSt7QBq5MSohKAA1KkAHmQ8GmqV-NwWj2eA3gAGsmn9QFCTPRKE1EIRKAA5chMH7_UBgpFPFFsQbkP6A4GgiHFJYrNaEDZbaYXUC0mp3X48R5SFEw-HwRG8lFopCYnF4ymE37Evn0Mmwcgc0DwQiwVDwaDxWDXBaCfLdQrdSKNaBsKrGmqJACCbUoVVpsBwvmihPAg0IpD6OE6tOQkFQ5vI8BwAHFyJVQDRPmiivrFeQYrxUoALFUAAHKdQmpwAxclm04BIOXzqcA22rFwApcsXALaKxcAp6apLw1K3FHB29qW-OCKDEPrJ1KkNVMzqpZBfQgNzugMUYh18WnVfU1aC4t2pRNSP2Lzku1dHaDjyeLyhK8CwtKAbgNAPOJm63ghDhHIAGVnZEqoI_ORwIhnTFRPQAKwFIe-qUPgkAAO6PjgTABsQMQxsQcYqoIe6EM-OCvrE4GIGwWy_s2t6EEwvj4FQ7axGw7TkKOhAWnU3wEVuiYxBwJLAQsABeMRCLSjE1Aut7LniaTrjet7Oq6aSoWJW7Hp-Z6EleMlHkwpR4fRhDsbU95Pi-bpxKAH5fj-oB_oBfEJuSLFsLwfJaTUXGgDxi4WYIAlbkJq6JkIyn6hJu6IPuvkLCAcmnhe172cUOnoZhBlGd-OC_gBQHIVOCRRqkTJsMFtTMaArGmLxk5ePklxAA). 
However the legend will reveal the ugly truth. 🤡

![image](https://github.com/apache/echarts/assets/13038071/a55a2d00-02d9-4a6e-8903-f64f0dfd4d66)

**Comment by fawdlstty** [comment]
Yes, this is the effect I want, and I hope Echarts can fix the problem and directly support it. The hack scheme is clearly not effective enough in its display

**Comment by fawdlstty** [comment]
a
